sympy/functions/special/gamma_functions.py

```diff
@@ -165,6 +163,8 @@ def _eval_conjugate(self):
 
     def _eval_is_real(self):
         x = self.args[0]
+        if intlike(x) and x <= 0:
+            return
```

Comment:
This should return `False` (can be checked simply with assumptions).
```py
        if x.is_positive or x.is_noninteger:
            return True
        if x.is_nonpositive and x.is_integer:
            return False
```

Reply:
@sidhantnagpal I think we should use `intlike` here:
```
>>> gamma(-1.0)
zoo
>>> a = gamma(-1.0, evaluate=False)
>>> a.args[0].is_integer
False
```
Also earlier `.is_real` returned `None` even though we could return `False`. I will return `False` now.

Reply:
If you want that change, why not just make it `x.is_nonpositive and intlike(x)`, ie, `intlike(x)` in place of `x.is_integer` in https://github.com/sympy/sympy/pull/16843#discussion_r291814685.

Reply:
@sidhantnagpal because `intlike` return `False` for symbols with integer assumption.
```python
In [8]: from sympy.core.compatibility import as_int                                                      

In [9]: def intlike(n): 
   ...:     try: 
   ...:         as_int(n, strict=False) 
   ...:         return True 
   ...:     except ValueError: 
   ...:         return False 
   ...:                                                                                                  

In [10]: x = symbols('x', integer=True)                                                                  

In [11]: intlike(x)                                                                                      
Out[11]: False
```

Reply:
OK, it should be fine then. Make sure to add test(s), both symbolic and numeric, for `is_real` checks covering the different scenarios.